### PR TITLE
Don't crash when package.json is corrupted

### PR DESCRIPTION
**Pull request by clochix**

### CODE REVIEW

**Comment by frankrousseau** [comment]
Could you add comments explaining why your code is added?

File: server/lib/app.coffee [hunk 1]
```diff
@@ -26,12 +26,22 @@ class exports.App
...
-                @app.server = start[start.length - 1]
-            else if fs.existsSync path.join(@app.dir, 'build', 'server.js')
-                @app.server = 'build/server.js'
-            else
-                @app.server = 'server.coffee'
+            manifestPath = path.join(@app.dir, "package.json")
+            serverPath   = path.join(@app.dir, 'build', 'server.js')
+            fs.readFileSync manifest, (err, data) ->
+                if err
+                    log.warn 'Error reading package.json'
```
Comment (nono) [inline server/lib/app.coffee]: I don't see where log is defined

File: server/lib/app.coffee [hunk 2]
```diff
@@ -26,12 +26,22 @@ class exports.App
...
+                else
+                    try
+                        manifest = JSON.parse data
+                    catch
+                        log.parse 'Error parsing package.json'
```
Comment (nono) [inline server/lib/app.coffee]: s/parse/error/ ?

File: server/lib/app.coffee [hunk 3]
```diff
@@ -26,12 +26,22 @@ class exports.App
...
-            if manifest.scripts?.start?
-                start = manifest.scripts.start.split(' ')
-                @app.server = start[start.length - 1]
-            else if fs.existsSync path.join(@app.dir, 'build', 'server.js')
-                @app.server = 'build/server.js'
-            else
-                @app.server = 'server.coffee'
+            manifestPath = path.join(@app.dir, "package.json")
+            serverPath   = path.join(@app.dir, 'build', 'server.js')
+            fs.readFileSync manifest, (err, data) ->
```
Comment (nono) [inline server/lib/app.coffee]: s/manifest/manifestPath/

Comment (nono) [inline server/lib/app.coffee]: and `readFileSync` doesn't take a callback

**Pull request closed without merging**